### PR TITLE
EZP-31477: Fixed loading Site Accesses on limitation list

### PR DESCRIPTION
**Pull request by mikadamczyk**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [EZP-31477](https://jira.ez.no/browse/EZP-31477)
| **Bug/Improvement**| no
| **New feature**    | yes
| **Target version** | `master`
| **BC breaks**      | no
| **Tests pass**     | yes
| **Doc needed**     | no
| **Related to**        | https://github.com/ezsystems/ezplatform-admin-ui/pull/1272

List of SiteAccesses on limitation list should be fetched from the service, not from configuration. We have an extension point which allows extending the list of SiteAccesses. To fetch all of SiteAccesses the siteAccessService should be used.

**QA**:
you need to check whether the limitations based on SiteAccesses continue to work

**TODO**:
- [x] Implement feature / fix a bug.
- [x] Implement tests.
- [x] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [x] Ask for Code Review.

### CODE REVIEW

File: eZ/Publish/Core/Limitation/SiteAccessLimitationType.php [hunk 1]
```diff
@@ -82,7 +81,7 @@ public function validate(APILimitationValue $limitationValue)
     {
         $validationErrors = [];
         foreach ($limitationValue->limitationValues as $key => $value) {
-            if (!isset($this->siteAccessList[$value])) {
+            if (!isset($this->getSiteAccessList()[$value])) {
```
Comment (alongosz) [inline eZ/Publish/Core/Limitation/SiteAccessLimitationType.php]: Please decrease its complexity back to `O(n)` by moving `$this->getSiteAccessList()` outside of the loop.

File: eZ/Publish/Core/settings/tests/common.yml [hunk 2]
```diff
@@ -60,3 +60,19 @@ services:
     ezpublish.siteaccess:
         class: eZ\Publish\Core\MVC\Symfony\SiteAccess
         arguments: ['default', 'uninitialized']
+
+    ezpublish.siteaccess_service:
+        class: eZ\Publish\Core\MVC\Symfony\SiteAccess\SiteAccessService
+        arguments:
+            - "@ezpublish.siteaccess.provider"
+            - "@ezpublish.config.resolver"
```
Comment (alongosz) [inline eZ/Publish/Core/settings/tests/common.yml]: CS:
```suggestion
            - '@ezpublish.siteaccess.provider'
            - '@ezpublish.config.resolver'
```

File: eZ/Publish/Core/Limitation/SiteAccessLimitationType.php (hunk 1 above)
Comment (adamwojs) [reply eZ/Publish/Core/Limitation/SiteAccessLimitationType.php]: `\eZ\Publish\Core\MVC\Symfony\SiteAccess\SiteAccessServiceInterface::exists` could be used instead. 

> Please decrease its complexity back to O(n) by moving $this->getSiteAccessList() outside of the loop.

What kind of complexity do you have in mind @alongosz ? :smirk:

**Comment by mikadamczyk** [comment]
@alongosz fixed, thanks for the vigilance!

**Pull request merged**